src/Overlay.tsx

```diff
@@ -0,0 +1,90 @@
+import styled from 'styled-components'
+import React, {ReactElement} from 'react'
+import {get, COMMON, POSITION} from './constants'
+import {ComponentProps} from './utils/types'
+import { useOverlay, AnchoredPositionHookSettings, TouchOrMouseEvent } from './hooks'
+import Portal from './Portal'
+
+type StyledOverlayProps = {
+  width?: keyof typeof widthMap
+  height?: keyof typeof heightMap
+  visibility: 'visible' | 'hidden'
+}
+
+const heightMap = {
+  'sm': '480px',
+  'md': '640px',
+  'auto': 'auto'
+}
+
+const widthMap = {
+  'sm': '256px',
+  'md': '320px',
+  'lg': '480px',
+  'xl': '640px',
+  'auto': 'auto'
+}
+
+/*TODO replace with shadow functional color variable when it's shipped to primer/primitives*/
+const StyledOverlay  = styled.div<StyledOverlayProps>`
```

Comment:
```suggestion
const StyledOverlay = styled.div<StyledOverlayProps>`
```

Reply:
Note: I noticed a few other formatting inconsistencies throughout. Make sure to run Prettier on all the edited files.

Reply:
just realized my prettier VSCode extension hasn't be auto formatting for who knows how long 😳